src/native/libs/build-local.sh

```diff
@@ -13,6 +13,27 @@
 SHIM_SOURCE_DIR="$1"/native/src
 INTERMEDIATE_OUTPUT_PATH="$2"
 TARGET_LIBRARY="$3"
+EMBED_ICU_DATA_PATH="$4"
+
+if [ -n "$EMBED_ICU_DATA_PATH" ]; then
+    header="${INTERMEDIATE_OUTPUT_PATH}embedded_icu_data.h"
+
+    cat <<EOF > "$header"
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifndef EMBEDDED_ICU_DATA
+#define EMBEDDED_ICU_DATA
+
+static unsigned char icu_data[] __attribute__((aligned(16))) = {
```

Comment:
Should this have `const` so that the ICU data are placed into a read-only segment?

Reply:
```suggestion
static const unsigned char icu_data[] __attribute__((aligned(16))) = {
```